src/Polyhedron.cu

```diff
@@ -93,24 +93,31 @@ __host__ __device__ bool does_edge_belong_to_face(SpacePoint a, SpacePoint b, co
 
 __host__ __device__ Face *find_face_next_to_edge(int vertex_id, Face *current_face, Polyhedron *polyhedron)
 {
+    const SpacePoint *point_a = current_face->get_vertices() + vertex_id;
+    const SpacePoint *point_b = point_a + 1; // By default it's just the next vertex
+    if(vertex_id + 1 == current_face->get_n_of_vertices()) // But if point_a was the last one
+        point_b = current_face->get_vertices(); // Then point b should be the first one
```

Comment:
Why not using the remainder of the division by `Face::n_of_vertices`, like you use it in `find_intersection_with_edge`? Is it just because you want to use the pointers to `SpacePoint`?

Reply:
I think you are correct and that way would be better

Reply:
Do you find
`const SpacePoint *point_b = current_face->get_vertices() + (vertex_id + 1) % current_face->get_n_of_vertices();`
better than the current one? Or do you see an even better way?

Reply:
I thought to do exactly like in another function, something like
`current_face->get_vertices()[(vertex_id + 1) % current_face->get_n_of_vertices()]` 
(maybe even not to create a new variable), but you're right, your solution is more elegant